src/measure_theory/measure/haar_lebesgue.lean

```diff
@@ -207,31 +290,27 @@ lemma map_linear_map_add_haar_eq_smul_add_haar
   {f : E →ₗ[ℝ] E} (hf : f.det ≠ 0) :
   measure.map f μ = ennreal.of_real (abs (f.det)⁻¹) • μ :=
 begin
+  lift f to E →L[ℝ] E using trivial,
   -- we reduce to the case of `E = ι → ℝ`, for which we have already proved the result using
   -- matrices in `map_linear_map_add_haar_pi_eq_smul_add_haar`.
   let ι := fin (finrank ℝ E),
   haveI : finite_dimensional ℝ (ι → ℝ) := by apply_instance,
   have : finrank ℝ E = finrank ℝ (ι → ℝ), by simp,
-  have e : E ≃ₗ[ℝ] ι → ℝ := linear_equiv.of_finrank_eq E (ι → ℝ) this,
-  -- next line is to avoid `g` getting reduced by `simp`.
-  obtain ⟨g, hg⟩ : ∃ g, g = (e : E →ₗ[ℝ] (ι → ℝ)).comp (f.comp (e.symm : (ι → ℝ) →ₗ[ℝ] E)) :=
-    ⟨_, rfl⟩,
-  have gdet : g.det = f.det, by { rw [hg], exact linear_map.det_conj f e },
-  rw ← gdet at hf ⊢,
-  have fg : f = (e.symm : (ι → ℝ) →ₗ[ℝ] E).comp (g.comp (e : E →ₗ[ℝ] (ι → ℝ))),
-  { ext x,
-    simp only [linear_equiv.coe_coe, function.comp_app, linear_map.coe_comp,
-      linear_equiv.symm_apply_apply, hg] },
-  simp only [fg, linear_equiv.coe_coe, linear_map.coe_comp],
-  have Ce : continuous e := (e : E →ₗ[ℝ] (ι → ℝ)).continuous_of_finite_dimensional,
-  have Cg : continuous g := linear_map.continuous_of_finite_dimensional g,
-  have Cesymm : continuous e.symm := (e.symm : (ι → ℝ) →ₗ[ℝ] E).continuous_of_finite_dimensional,
-  rw [← map_map Cesymm.measurable (Cg.comp Ce).measurable, ← map_map Cg.measurable Ce.measurable],
-  haveI : is_add_haar_measure (map e μ) := (e : E ≃+ (ι → ℝ)).is_add_haar_measure_map μ Ce Cesymm,
-  have ecomp : (e.symm) ∘ e = id,
-    by { ext x, simp only [id.def, function.comp_app, linear_equiv.symm_apply_apply] },
-  rw [map_linear_map_add_haar_pi_eq_smul_add_haar hf (map e μ), measure.map_smul,
-    map_map Cesymm.measurable Ce.measurable, ecomp, measure.map_id]
+  have e : E ≃L[ℝ] ι → ℝ := continuous_linear_equiv.of_finrank_eq this,
+  set g := (e : E →L[ℝ] (ι → ℝ)).comp (f.comp (e.symm : (ι → ℝ) →L[ℝ] E)) with hg,
+  have gdet : g.det = f.det := by exact linear_map.det_conj _ _,
+  rw [←continuous_linear_map.symm_comp_eq, continuous_linear_map.eq_comp_symm] at hg,
+  clear_value g,
+  subst hg,
+  rw [←continuous_linear_map.det, ←gdet] at ⊢ hf,
+  simp only [continuous_linear_map.coe_comp, linear_map.coe_comp, continuous_linear_map.coe_coe,
+    continuous_linear_equiv.coe_coe],
+  have me : measurable e := e.continuous.measurable,
+  have mg : measurable g := g.continuous.measurable,
+  have mesymm : measurable e.symm := e.symm.continuous.measurable,
+  rw [← map_map mesymm (mg.comp me), ← map_map mg me, ←g.coe_coe,
+    map_linear_map_add_haar_pi_eq_smul_add_haar hf (map e μ), measure.map_smul, map_map mesymm me,
+    e.symm_comp_self, measure.map_id],
```

Comment:
I must be a bit over the limit because it doesn't time out in local.

Note that this PR isn't ready yet as I need to PR the preliminary material separately.